bin/release.py

```diff
@@ -0,0 +1,125 @@
+"""A script which triggers and wait for an azure devops to complete a release"""
+from ast import parse
+import os
+import sys
+import time
+from azure.devops.connection import Connection
+from msrest.authentication import BasicAuthentication
+
+# https://learn.microsoft.com/en-us/rest/api/azure/devops/release/releases/get-release?view=azure-devops-rest-6.0&tabs=HTTP#environmentstatus
+FAILED_STATUSES = ["canceled", "partiallySucceeded", "rejected"]
+COMPLETED_STATUSES = ["succeeded", "canceled", "partiallySucceeded", "rejected"]
+
+
+def create_ado_connection(organization, ADO_PAT) -> Connection:
+    """Returns an ADO connection to call the ADO Rest APIs."""
+
+    organization_url = "https://dev.azure.com/{0}".format(organization)
+    credentials = BasicAuthentication("", ADO_PAT)
+    connection = Connection(base_url=organization_url, creds=credentials)
+
+    return connection
+
+
+def get_release_definition(project, pipeline_name, release_client):
+    """Returns the ADO definition for the given pipeline"""
+    # Get all release definitions in a given project
+    project_release_definitions = release_client.get_release_definitions(project)
+
+    # Filter release definitions with the given pipeline name
+    pipeline_release_definitions = []
+    for definition in project_release_definitions.value:
+        if definition.name == pipeline_name:
+            pipeline_release_definitions.append(definition)
+
+    if pipeline_release_definitions is None or len(pipeline_release_definitions) == 0:

```

Comment:
Python will allow you to handle both of these conditions in a single check by writing
```suggestion
    if not pipeline_release_definitions:
```
Both `None` and a `list` of length `0` are considered ["falsy"](https://docs.python.org/3/library/stdtypes.html#truth-value-testing).

More importantly though, it's not actually possible for `pipeline_release_definitions` to be `None`. You defined it just above. It will always be a `[]` in the worst case. Was this just a cautious C#-style check? 🤔